zkvm/src/types.rs

```diff
@@ -140,6 +142,12 @@ pub struct FrozenValue {
     pub(crate) flv: Commitment,
 }
 
+#[derive(Copy, Clone, Debug)]
+pub enum ScalarKind {
```

Comment:
`ScalarKind` sounds meta to me: like, a type of a type. Maybe `ScalarWitness` is more apt and consistent with the rest of the code? It's a "real witness value for the given scalar representation".

```suggestion
pub enum ScalarWitness {
```